src/raftstore/store/store.rs

```diff
@@ -1789,7 +1795,63 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         };
     }
 
+    fn poll_compaction_event(&mut self) {
+        if self.compaction_listener.is_none() {
+            return;
+        }
+
+        let mut events = vec![];
+        loop {
+            match self.compaction_listener.as_ref().unwrap().try_recv() {
+                Ok(event) => events.push(event),
+                Err(TryRecvError::Empty) => break,
+                Err(e) => panic!("unexpected error {:?}", e),
+            }
+        }
+
+        for event in events.drain(..) {
+            // When calculating region size, we only consider write and default
+            // column families.
+            if event.cf != CF_WRITE && event.cf != CF_DEFAULT {
+                continue;
+            }
+            // Compactions in level 0 and level 1 are very frequently.
+            if event.output_level < 2 {
+                continue;
+            }
+
+            let mut influenced_regions = vec![];
+            for (_, region_id) in self.region_ranges
+                .range((Included(event.start_key), Included(event.end_key)))
+            {
+                influenced_regions.push(region_id);
+            }
+
+            let output_level_str = event.output_level.to_string();
+            COMPACTION_RELATED_REGION_COUNT
+                .with_label_values(&[&output_level_str])
+                .observe(influenced_regions.len() as f64);
+
+            if influenced_regions.is_empty() {
+                continue;
+            }
+            let region_estimated_change = event.total_size / influenced_regions.len();
+            let score = cmp::max(
```

Comment:
Any particular reason that we need to calculate a score like this?
Can we just adjust the `size_diff_hint` of the peer?